File: src/main/java/com/dansoftware/libraryapp/gui/updatedialog/segment/download/DownloadSegment.java
package com.dansoftware.libraryapp.gui.updatedialog.segment.download;

import com.dansoftware.libraryapp.gui.context.Context;
import com.dansoftware.libraryapp.gui.util.ImprovedFXMLLoader;
import com.dansoftware.libraryapp.i18n.I18N;
import com.dansoftware.libraryapp.update.UpdateInformation;
import com.dansoftware.sgmdialog.TitledSegment;
import javafx.scene.Node;
import org.jetbrains.annotations.NotNull;

/**
 * Allows the user to download the new app from the internet in a {@link com.dansoftware.libraryapp.gui.updatedialog.UpdateDialog}.
 * The user can select what type of file should be downloaded.
 *
 * @author Daniel Gyorffy
 */
public class DownloadSegment extends TitledSegment {

    private final Context context;
    private final UpdateInformation updateInformation;

    public DownloadSegment(@NotNull Context context, @NotNull UpdateInformation updateInformation) {
        super(I18N.getValues().getString("segment.download.name"),
                I18N.getValues().getString("segment.download.title"));
        this.context = context;
        this.updateInformation = updateInformation;
    }


    @Override
    protected Node getCenterContent() {
        return new ImprovedFXMLLoader(
                new DownloadSegmentController(context, updateInformation),
                getClass().getResource("DownloadSegment.fxml"),
                I18N.getValues()).load();
    }
}

File: src/main/java/com/dansoftware/boomega/gui/updatedialog/segment/download/DownloadSegment.kt
/*
 * Boomega
 * Copyright (C)  2021  Daniel Gyoerffy
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.dansoftware.boomega.gui.updatedialog.segment.download

import com.dansoftware.boomega.gui.api.Context
import com.dansoftware.boomega.i18n.I18N
import com.dansoftware.boomega.update.Release
import com.dansoftware.sgmdialog.TitledSegment
import javafx.scene.Node

/**
 * Allows the user to download the new app from the internet in a [com.dansoftware.boomega.gui.updatedialog.UpdateDialog].
 * The user can select what type of file should be downloaded.
 *
 * @author Daniel Gyorffy
 */
class DownloadSegment(
    private val context: Context,
    private val release: Release
) : TitledSegment(
    I18N.getValue("update.segment.download.name"),
    I18N.getValue("update.segment.download.title")
) {
    override fun getCenterContent(): Node = DownloadView(context, release)
}